presto-main/src/main/java/com/facebook/presto/operator/aggregation/MaxAggregation.java

```diff
@@ -0,0 +1,397 @@
+        {
+            return new SingleLongState(type, methodHandle);
+        }
+
+        @Override
+        public Class<? extends LongState> getSingleStateClass()
+        {
+            return SingleLongState.class;
+        }
+
+        @Override
+        public LongState createGroupedState()
+        {
+            return new GroupedLongState(type, methodHandle);
+        }
+
+        @Override
+        public Class<? extends LongState> getGroupedStateClass()
+        {
+            return GroupedLongState.class;
+        }
+
+        public static class GroupedLongState
+                extends AbstractGroupedAccumulatorState
+                implements LongState
+        {
+            private final Type type;
+            private final MethodHandle methodHandle;
+            private final LongBigArray values = new LongBigArray();
+            private final BooleanBigArray nulls = new BooleanBigArray();
+
+            public GroupedLongState(Type type, MethodHandle methodHandle)
+            {
+                this.type = type;
+                this.methodHandle = methodHandle;
+            }
+
+            @Override
+            public void ensureCapacity(long size)
+            {
+                values.ensureCapacity(size);
+                nulls.ensureCapacity(size);
+            }
+
+            @Override
+            public long getEstimatedSize()
+            {
+                return values.sizeOf() + nulls.sizeOf();
+            }
+
+            @Override
+            public Type getType()
+            {
+                return type;
+            }
+
+            @Override
+            public MethodHandle getMethodHandle()
+            {
+                return methodHandle;
+            }
+
+            @Override
+            public long getLong()
+            {
+                return values.get(getGroupId());
+            }
+
+            @Override
+            public void setLong(long value)
+            {
+                values.set(getGroupId(), value);
+            }
+
+            @Override
+            public boolean isNull()
+            {
+                return nulls.get(getGroupId());
+            }
+
+            @Override
+            public void setNull(boolean value)
+            {
+                nulls.set(getGroupId(), value);
+            }
+        }
+
+        public static class SingleLongState
+                implements LongState
+        {
+            private final Type type;
+            private final MethodHandle methodHandle;
+            private long value;
+            private boolean nul;
+
+            public SingleLongState(Type type, MethodHandle methodHandle)
+            {
+                this.type = type;
+                this.methodHandle = methodHandle;
+                this.value = 0;
+                this.nul = true;
+            }
+
+            @Override
+            public long getEstimatedSize()
+            {
+                return 16;
```

Comment:
Use ClassLayout.parseClass(SingleLongMinMaxState.class).instanceSize(), and cache it in a static field